### PR TITLE
[#5169] Block switching indicators to cumulative reporting

### LINKED ISSUE #5169: Feature Request: Block switching indicators to cumulative reporting
### What are you trying to do?

Cumulative reporting has an impact on how period updates are aggregated and how actual period values are attained.

Switching between the two when period updates already exist can make the calculations complicated and error prone. 

### Describe the solution you'd like

Deny changing the reporting method after a period update has been made.

### Have you consider alternatives?

_No response_

### Frontend

 - [ ] Call backend API to check if any period updates have been made on indicator
 - [ ] Hide toggle for reporting method of indicator

### Backend

 - [ ] Provide endpoint (or modify existing endpoint) to return the number of period indicator updates
 - [ ] Reject attempts to change reporting method on endpoint if period indicator updates exist

## Design

???

### PR DESCRIPTION
**Pull request by zuhdil**
# TODO / Done

Summarize what has been changed / what has to be done in order to finalize the PR.

 - [x] Endpoint to check if any period updates have been made on the indicator
 - [x] Validate cumulative attribute update
 - [x] Disable/enable cumulative switch for indicators on project editor

# Test plan

What tests are necessary to ensure this works or doesn't break anything working

 - [x] Unit test
 - [ ] Manual test

---

Closes #5169

### CODE REVIEW

File: akvo/rest/serializers/indicator.py [hunk 1]
```diff
@@ -103,6 +104,15 @@ def update(self, instance, validated_data):
         create_or_update_disaggregation_targets(instance, disaggregation_targets)
         return instance
 
+    def validate_cumulative(self, data):
+        if not self.instance or self.instance.cumulative == data:
+            return data
+        if get_indicator_contribution_count(self.instance) > 0:
+            raise serializers.ValidationError(
+                "Switching cumulative reporting is not allowed when indicator already have updates"
```
Comment (MichaelAkvo) [inline akvo/rest/serializers/indicator.py:112]:
```suggestion
                "Switching cumulative reporting is not allowed when the indicator already has updates"
```

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]
Minor changes required @zuhdil

File: akvo/rsr/models/result/indicator.py [hunk 2]
```diff
@@ -180,6 +180,18 @@ def delete(self, *args, **kwargs):
...
+        family = {self.pk}
+        search_depth = 0
+        while depth is None or search_depth < depth:
+            children = Indicator.objects.filter(parent_indicator__in=family).values_list('pk', flat=True)
+            if family.union(children) == family:
+                break
+
+            family = family.union(children)
+            search_depth += 1
+        return Indicator.objects.filter(pk__in=family)
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply akvo/rsr/models/result/indicator.py]: Fixed

**Review by zuhdil — COMMENTED** [review]
The Frontend looks good to me.

**Review by MichaelAkvo — APPROVED** [review]
Small bug found, but we can merge it as it is and handle that later

**Pull request merged**